### PR TITLE
[AERIE 2131] Database Migration Script

### PR DESCRIPTION
**Pull request by Mythicaeda**
* **Tickets addressed:** AERIE-2118, AERIE-2131
* **Review:** By commit  
* **Merge strategy:** Merge (no squash)  

## Description

Finishes the python CLI tool began in #360.

The script now:

- avoids displaying databases that don't have migrations
- avoids attempting to apply/revert changes that have already been applied
- prevents the user from skipping a migration when in step-by-step
- has the user start the tool in either `apply` or `revert` mode, rather than allowing them to choose per-migration-step
- no longer raises an exception if not given an integer on the select database screen
- allows the user to select a list of databases to migrate, rather than either one or all
- displays an error in the event that a user attempts to migrate a database when they are missing some of the migration steps that exist on the server

For more information on the layout of the `up.sql`/`down.sql`, see #360. 

## Verification

Ran the script against the migrations found in `deployment/sample-migrations` and verified that they ran both up and down without issues. (Note: `test_migration` in AerieMerlin requires two plans to exist prior to being able to apply it, as it's a demonstration that we can insert/delete/modify data in migrations, rather than a schema change)

## Documentation

* Added a new page to `Upgrade Guides`, `Aerie Database Migration Script`.
* Updated the index page for `Upgrade Guides` to display the Table of Contents.

### CODE REVIEW

File: deployment/aerie_db_migration [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+      to_migrate_set.discard(db)
+
+if to_migrate_set:
+  print("\033[91mError\033[0m: The following Database(s) do not contain migrations:\n\t"
+                          +"\n\t".join(to_migrate_set))
+  sys.exit()
+
+if not migration_db:
+  print("\033[91mError\033[0m: No database migrations found.")
+  sys.exit()
```
Comment (pcrosemurgy) [inline deployment/aerie_db_migration]: I'd make this a `sys.exit(1)` to indicate that the process failed to the shell.

Comment (pcrosemurgy) [reply deployment/aerie_db_migration]: This applies to all the other exits as well :)

Comment (Mythicaeda) [reply deployment/aerie_db_migration]: Done, although I left the `sys.exit()` following the user entering a quit key untouched ;)

Comment (pcrosemurgy) [reply deployment/aerie_db_migration]: Thanks! Good call

**Review by camargo — APPROVED** [review]

**Review by pcrosemurgy — APPROVED** [review]
Nice work! This looks good to me.

**Pull request merged**